slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java

```diff
@@ -0,0 +1,301 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class SLangConverter {
+
+  public Tree parse(String slangFile) throws IOException {
+    SLangLexer lexer = new SLangLexer(CharStreams.fromFileName(slangFile));
+    CommonTokenStream tokens = new CommonTokenStream(lexer);
+    SLangParser parser = new SLangParser(tokens);
+
+    SLangParseTreeVisitor slangVisitor = new SLangParseTreeVisitor();
+    return slangVisitor.visit(parser.slangFile());
+  }
+
+
+  private static class SLangParseTreeVisitor extends SLangBaseVisitor<Tree> {
+
+    @Override
+    public Tree visitSlangFile(SLangParser.SlangFileContext ctx) {
+      List<Tree> children = ctx.typeDeclaration()
+          .stream()
+          .map(type -> visit(type))
+          .collect(toList());
+
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitNativeExpression(SLangParser.NativeExpressionContext ctx) {
+      List<Tree> children = ctx.nativeBlock()
+              .stream()
+              .map(nativeBlockContext -> visit(nativeBlockContext))
+              .collect(toList());
+
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitStatementOrExpression(SLangParser.StatementOrExpressionContext ctx) {
+      List<Tree> children = ctx.disjunction()
+              .stream()
+              .map(disjunctionContext -> visit(disjunctionContext))
+              .collect(toList());
+
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitMethodDeclaration(SLangParser.MethodDeclarationContext ctx) {
+      return visit(ctx.methodBody());
+    }
+
+    @Override
+    public Tree visitMethodBody(SLangParser.MethodBodyContext ctx) {
+      if (ctx.SEMICOLON() != null) {
+        return new NativeTreeImpl(null, new SNativeKind(), new ArrayList<>());
+      }
+      return visit(ctx.block());
+    }
+
+    @Override
+    public Tree visitBlock(SLangParser.BlockContext ctx) {
+      List<Tree> children = ctx.statementOrExpression()
+              .stream()
+              .map(statementOrExpressionContext -> visit(statementOrExpressionContext))
+              .collect(toList());
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitNativeBlock(SLangParser.NativeBlockContext ctx) {
+      List<Tree> children = ctx.statementOrExpression()
+              .stream()
+              .map(statementOrExpressionContext -> visit(statementOrExpressionContext))
+              .collect(toList());
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitDisjunction(SLangParser.DisjunctionContext ctx) {
+      int conjunctionSize = ctx.conjunction().size();
+
+      if (conjunctionSize > 1) {
+        if (conjunctionSize > 2) {
```

Comment:
When > 1, it should always return `BinaryExpressionTree` object, with nested nested children `BinaryExpressionTree` depending on the number of children